src/ecdsa/ecdh.py

```diff
@@ -97,39 +108,123 @@ def load_private_key(self, private_key):
+        Curve is get from PEM and check if it the same as current curve
+
+        Note, the only PEM format supported is the RFC5915
+        Look at keys.py:SigningKey.from_pem()
+        it needs to have `EC PRIVATE KEY` section
+
+        :param private_key: text with PEM-encoded private ECDSA key
+        :return: public (verifying) key from this private key
+        """
         return self.load_private_key(SigningKey.from_pem(private_key))
 
     def get_public_key(self):
+        """
+        Get public key from local private key.
+        Usually needs to send it to remote party.
+        :return: public (verifying) key from local private key
+        """
         return self.private_key.get_verifying_key()
 
     def load_received_public_key(self, public_key):
+        """
+        Load public key from VerifyingKey (keys.py) class.
+        Needs to have the same curve as set as current for ecdh operation.
+        If curve is not set - it sets it from VerifyingKey
+        :param public_key: Initialised VerifyingKey class
+        :return:
+        """
         if self.curve is None:
             self.curve = public_key.curve
         if self.curve != public_key.curve:
             raise InvalidCurveError("Curve mismatch.")
         self.public_key = public_key
 
     def load_received_public_key_bytes(self, public_key_str):
+        """
+        Load public key from bytes string.
+        Uses current curve and checks if key length corresponds to
+        the current curve.
+        :param public_key_str: public key in bytes string format
+        :return: none
+        """
         return self.load_received_public_key(
             VerifyingKey.from_string(public_key_str, self.curve))
 
     def load_received_public_key_der(self, public_key_str):
+        """
+        Load public key from DER binary string.
+        Curve is get from DER and check if it the same as current curve
+
+        Note, the only DER format supported is the RFC5912
+        Look at keys.py:VerifyingKey.from_der()
+
+        :param public_key_str: binary string with the DER encoding of public ECDSA key
+        :return: none
+        """
         return self.load_received_public_key(VerifyingKey.from_der(public_key_str))
 
     def load_received_public_key_pem(self, public_key_str):
+        """
+        Load public key from PEM string.
+        Curve is get from PEM and check if it the same as current curve
+
+        Note, the only PEM format supported is the RFC5912
+        Look at keys.py:VerifyingKey.from_pem()
+
+        :param public_key_str: text with PEM-encoded public ECDSA key
+        :return: none
+        """
         return self.load_received_public_key(VerifyingKey.from_pem(public_key_str))
 
     def generate_sharedsecret_bytes(self):
+        """
+        Generate shared secret from local private key and remote public key.
+        Needs to have initialized private key and received public key before
+        generating.
```

Comment:
"The objects needs to have both private key and received public key before generation is allowed"?